crates/pallet-domains/src/lib.rs

```diff
@@ -174,9 +191,21 @@ mod pallet {
             Ok(())
         }
 
-        // TODO: proper weight
         #[pallet::call_index(1)]
-        #[pallet::weight((10_000, Pays::No))]
+        #[pallet::weight(
+            match fraud_proof {
+                FraudProof::InvalidStateTransition(proof) => (
+                    if proof.domain_id.is_system() {
+                        T::WeightInfo::submit_system_domain_invalid_state_transition_proof()
+                    } else {
+                        T::WeightInfo::submit_core_domain_invalid_state_transition_proof()
+                    },
```

Comment:
Yeah, but the core domain fraud-proof is still a valid extrinsic to pallet-domains and will be included in the primary block, thus we still need a weight for the core domain fraud-proof here.

It is also because the core domain fraud-proof is not fully handled here as the system domain, thus we should not treat the weight of the core domain fraud-proof as the same as the system domain fraud-proof, and `submit_core_domain_invalid_state_transition_proof` just reflect the real weight that core domain fraud-proof should have. This should also answer the `submit_bundle` question.

Reply:
Right, we can reject the core domain fraud-proof directly and won't need the weight here. But we may still need the core domain bundle weight as the core domain bundle is expected to stored at the primary chain and can't be rejected.

Reply:
Yeah, that makes sense, resolved another comment.